### DISCUSSION TITLE
Docker opendcim/opendcim  23.01

### DISCUSSION THREAD

**Discussion by coreyman2000** [opendcim/openDCIM#1395] [Q&A] [answered]
Getting this error when trying to log in with default dcim user.

The ldap_debug_password is set and only the user dcim is valid for recovery. Run UPDATE fac_Config SET Value='' WHERE Parameter='LDAP_Debug_Password'; to clear the debug password.

here is my docker compose 

version: '3'
services:
  opendcim:
    image: opendcim/opendcim:21.01
    ports:
      - 80:80
    depends_on:
      - db
    environment:
      - OPENDCIM_DB_HOST=db
      - OPENDCIM_DB_NAME=opendcim
      - OPENDCIM_DB_USER=root
      - OPENDCIM_DB_PASS=your_mysql_password
      - OPENDCIM_AUTH=LDAP

  db:
    image: mysql:5.7
    volumes:
      - ./db_data:/var/lib/mysql
    environment:
      - MYSQL_ROOT_PASSWORD=your_mysql_password
      - MYSQL_DATABASE=opendcim

**Comment by LFrank2021** [comment]
Pulling 21.01 is intentional?
Is moist likely not the solution to your problem, but it is a discrepancy.
Did you try dcim/dcim?
I think I heard as much in the according YouTube video.

**Comment by coreyman2000** [comment]
looks like i was trying different versions i tired both, on 23.01 now and yes no matter the account dcim:dcimpassword 
or dcim/dcim both show the same error, i deleted the DB and source files again and still no luck :( 

The ldap_debug_password is set and only the user dcim is valid for recovery. Run UPDATE fac_Config SET Value='' WHERE Parameter='LDAP_Debug_Password'; to clear the debug password.

i also tried on a different docker host and same error :(

**Comment by coreyman2000** [comment]
ran wireshark and noticed
Expires: Thu, 19 Nov 1981 08:52:00 GMT, why would this date bad is there something wrong with apace or my local chrome? 
username=dcim&password=dcimpassword&submit=SubmitHTTP/1.1 200 OK
Date: Tue, 11 Jul 2023 15:19:17 GMT
Server: Apache/2.4.52 (Ubuntu)
Set-Cookie: PHPSESSID=pe17llnc7brllm57337jiv3paf; path=/; secure; HttpOnly
Expires: Thu, 19 Nov 1981 08:52:00 GMT
Cache-Control: no-store, no-cache, must-revalidate
Pragma: no-cache
Vary: Accept-Encoding
Content-Encoding: gzip
Content-Length: 4270
Keep-Alive: timeout=5, max=100
Connection: Keep-Alive
Content-Type: text/html; charset=UTF-8

**Comment by coreyman2000** [comment]
turns our password is dcimadmin however still just returns to login screeen :(

**Comment by samilliken** [comment]
I am struggling to understand what you are actually trying to ask here.   Is this a new db or a migration from an existing one?

**Comment by bhoglo** [comment]
@samilliken I am actually having the same issue described by @coreyman2000. It's a fresh install of mariaDB 11 and openDCIM 23.01 containers. It proceeds through the install.php->container-install.php process initially, and then brings you to a login screen. I've verified the database is actually being populated. If you attempt to use anything other than the dcim account with the LDAP_Debug_Password, you get the error mentioned above (expected). If you use the dcim/dcimadmin login, it just spits you back out at the login screen with no errors (not expected). I am not able to login and view the actual site. I've tried clearing the debug password, but that didn't change any outcome (as expected). Below is the screenshot of all I can access.

![Screenshot_2023-07-21_08-31-35](https://github.com/opendcim/openDCIM/assets/5169910/60c2d9f8-8b26-4445-b5dc-62bb24534565)

**Comment by samilliken** [comment]
What environment variables are you passing to the container?

 

Scott

**Reply by bhoglo** [reply]
environment:
      - OPENDCIM_DB_HOST=localhost
      - OPENDCIM_DB_PORT=3306
      - OPENDCIM_DB_NAME=dcim
      - OPENDCIM_DB_USER=dcim
      - OPENDCIM_DB_PASS=Xit4PoduKid5Mifp
      - OPENDCIM_AUTH=LDAP
      - TZ=America/Chicago

**Comment by samilliken** [comment]
So after you login with the ldap debug password, you can’t get to the Configuration screen to put in your actual LDAP configuration?

 

Scott

**Reply by bhoglo** [reply]
That is correct. It will not bring me to the configuration screen. It will POST to login_ldap.php -> redirect to index.php -> then redirect to login_ldap.php.

**Comment by samilliken** [comment]
I just tested with a fresh, empty db on 23.01.

 

Logged in as dcim/dcimadmin

Was able to click on Edit Configuration

Then I could setup an actual authentication scheme (in my case, OIDC)

Saved.   Shut down container and restarted with OPENDCIM_AUTH=OIDC

 

So I think you’re not doing it right.   Container is functioning as intended.

 

That being said, use 23.02 instead of 23.01, because it has several bug fixes in it.

 

Scott

**Reply by coreyman2000** [reply]
Mine just go back to the login screen, :( don't see any edit , did the exact same steps ,I can see the db is populated

**Comment by samilliken** [comment]
The only way that I was able to replicate anything close to what you stated was by running the container and trying to directly interact with it.   You can't do that - it needs to go through a reverse proxy with HTTPS enabled.   I have changed the notes on the container from:

SSL is not enabled in the container image. It is recommended that you run through a reverse proxy, such as nginx, for providing a secure connection to the application.

To now say that it is REQUIRED to run through a reverse proxy or load balancer.

**Reply by bhoglo** [reply]
Thanks @samilliken. I was able to get into the configuration page on a fresh install of Centos 7 (without a reverse proxy). Issue is on my end, thanks for looking into it!

**Reply by coreyman2000** [reply]
What was the fix? I'm running Ubuntu and docker, and even have same issues in rocky linux

**Reply by bhoglo** [reply]
I made no changes to my configuration. I only spun up a brand new VM on Cent OS 7. Worked first try.

**Reply by samilliken** [reply, marked as answer]
@coreyman2000 the answer is that the container works fine.   If you are not running behind a reverse proxy/loadbalancer that adds SSL, it's your setup that needs to be changed.

**Reply by coreyman2000** [reply]
thanks that was it started working right away soon as i created a ngix proxy for ssl
